### ISSUE TITLE
Failer on windows when pid is from dead process

### ISSUE THREAD

**Issue by xiegeo** [nightlyone/lockfile#10]
os: windows (tested on win10)

Reproduce:
1. Create lock file.
2. Exit process without cleaning up lock file. (as happens when you crash)
3. TryLock from new process.

This will produce `OpenProcess: The parameter is incorrect.` error, when it should succeed.

Cause:
```
func (l Lockfile) GetOwner() (*os.Process, error) {
	...
		p, err := os.FindProcess(pid)
		if err != nil {
			return nil, err
		}
	...
}
```

https://golang.org/pkg/os/#FindProcess

> On Unix systems, FindProcess always succeeds and returns a Process for the given pid, regardless of whether the process exists.  

But it's not documented for windows, and appears to return the error when process does not exit.

Fix:

We can assume that if the process exists, then FindProcess always succeeds in any operating system. There for, if FindProcess process returns an error, the process does not exist. So instead of returning the err, return ErrInvalidPid or ErrDeadOwner sould work here.

**Comment by nightlyone** [comment]
@xiegeo Would you be interested in creating a pull-request to fix this? I am not a Windows expert and thus have trouble fixing and reproducing this.

**Comment by stewi1014** [comment]
I can write a fix for it. I'll get working on it now.

**Comment by stewi1014** [comment]
I've just found out that on windows, checking if a dead process is alive will fail at os.FindProcess. Looks like it's not going to be a simple fix of the isProcessAlive function like I thought it would.

**Comment by stewi1014** [comment]
Ok, I think I've finished making a windows patch.
https://github.com/stewi1014/lockfile-nightlyone

If someone could test it on linux (I'm not at home with my Linux box atm) and check I haven't broken it on Linux, I'll start a pull request. Else I'll simply check it when I get home and start one then.

**Comment by xiegeo** [comment]
@stewi1014 I don't see any changes outside of lockfile_test.go.

**Comment by stewi1014** [comment]
@xiegeo Ah yeah, I'm dumb. I forgot to upload the changes. I'll fix that now.

**Comment by xiegeo** [comment]
@stewi1014 That a lot of changes. Behavior-wise, the only thing changed is added a check for `if uintptr(scerr) == 87` for windows, right?

I have changed my own copy and usage to allow the program to proceed on any unexpected error. Allow two instances to run at the same time is far less harmful then allowing none to run for my use case (wastes bandwidth and filling up the error log, vs sending someone on location)

I also looked at the original code and wonders what happens if old pid from dead process is then reassigned to a new unrelated process. In http://www.man7.org/tlpi/code/online/dist/filelock/create_pid_file.c.html a file descriptor is returned and old process is assumed dead if the pid file is not locked.

**Comment by stewi1014** [comment]
@xiegeo Yeh, I made a lot of changes. My personal opinion is that the library at large needs a rewrite, which is why I wrote my own library rather than bothering this community with a bunch of large changes that may or may not be welcome.

I'm afraid I have no idea what you're talking about in the second paragraph, but If I understand it correctly that you're making your own fixes instead, then that's cool. I'm not particularly fussed if my changes are deemed too radical. I mostly copied and pasted the process running check function from my own library and made changes to the existing library to use it.
